components/editableText/EditableText.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { SmallButton, Icon, Input } from 'react-components';
+import { c } from 'ttag';
+
+const EditableText = ({ icon, onSubmit, initialText, children, readOnly, className, ...rest }) => {
+    const [inputValue, setInputValue] = useState(initialText);
+    const [editing, setEditing] = useState(false);
+
+    useEffect(() => {
+        if (editing) {
+            setInputValue(initialText);
+        }
+    }, [editing, initialText]);
+
+    const onToggleEditing = () => {
+        setEditing(!editing);
+    };
+
+    const submit = (value) => {
+        onSubmit(value);
+        setEditing(false);
+    };
+
+    const handleFormSubmit = (e) => {
+        e.preventDefault();
```

Comment:
Is it really needed?

Reply:
This prevents form submission from happening. Essentially, it's not needed, because it would be prevented anyway, but with a warning thrown to console: `Form submission canceled because the form is not connected`